DevIncubator_Autopark/AllCollections/MyStack.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace DevIncubator_Autopark.AllCollections
+{
+    public class MyStack<T> : IEnumerable<T>
+    {
+        public int Count => GetCount();
+
+        private T[] _stack;
+        private int _headItemIndex;
+
+        public MyStack()
+        {
+            int defaultCapacity = 10;
+            _stack = new T[defaultCapacity];
+
+            _headItemIndex = -1;
+        }
+
+        public MyStack(IEnumerable<T> stack)
+        {
+            if (stack == null)
+            {
+                throw new ArgumentNullException(nameof(stack), "The transmitted data cannot be null");
+            }
+
+            foreach (var item in stack)
+            {
+                Push(item);
+            }
+        }
+
+        public void Push(T item)
+        {
+            int resizeValue = 2;
+
+            if (Count == _stack.Length - 1)
+            {
+                var resizeStack = new T[_stack.Length * resizeValue];
```

Comment:
Well it is bad idea because in this case you can spend too much memory when stack will contain too many elements (billion for example). So better to use addition for resizing of stack.